File: VB/WebDashboard_EFDataSource/Models/NorthwindDataModel.vb
Imports System.Data.Entity

Namespace WebDashboard_EFDataSource.Models

    Partial Public Class NorthwindDataModel
        Inherits DbContext

        Public Sub New()
            MyBase.New("name=NWindConnectionString")
        End Sub

        Public Overridable Property SalesPersons() As DbSet(Of SalesPerson)

        Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
            modelBuilder.Entity(Of SalesPerson)().Property(Function(e) e.UnitPrice).HasPrecision(10, 4)

            modelBuilder.Entity(Of SalesPerson)().Property(Function(e) e.Extended_Price).HasPrecision(19, 4)
        End Sub
    End Class
End Namespace

File: CS/WebDashboard_EFDataSource/Models/NorthwindDataModel.cs
namespace WebDashboard_EFDataSource.Models {
    using System.Data.Entity;

    public partial class NorthwindDataModel : DbContext {
        public NorthwindDataModel()
            : base("name=NWindConnectionString") {
        }

        public virtual DbSet<SalesPerson> SalesPersons { get; set; }

        protected override void OnModelCreating(DbModelBuilder modelBuilder) {
            modelBuilder.Entity<SalesPerson>()
                .Property(e => e.UnitPrice)
                .HasPrecision(10, 4);

            modelBuilder.Entity<SalesPerson>()
                .Property(e => e.Extended_Price)
                .HasPrecision(19, 4);
        }
    }
}
